File: Capstone-Project-dev/Capstone-Project-dev/fe/mobile/android/app/src/main/java/com/capstone/nawaco/bridge/media/MediaBridgeModule.java
package com.capstone.nawaco.bridge.media;

import androidx.annotation.NonNull;

import com.capstone.nawaco.domain.repository.MediaRepository;
import com.capstone.nawaco.infrastructure.security.PermissionManager;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

import java.io.File;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class MediaBridgeModule extends ReactContextBaseJavaModule {
    private final MediaRepository mediaRepository;
    private final PermissionManager permissionManager;
    private final ExecutorService executorService;

    public MediaBridgeModule(ReactApplicationContext reactContext,
                             MediaRepository mediaRepository,
                             PermissionManager permissionManager) {
        super(reactContext);
        this.mediaRepository = mediaRepository;
        this.permissionManager = permissionManager;
        this.executorService = Executors.newFixedThreadPool(4);
    }

    @NonNull
    @Override
    public String getName() {
        return "MediaModule";
    }

    @ReactMethod
    public void uploadCapturedImage(final String filePath, final Promise promise) {
        if (!permissionManager.canAccessFullFeatures()) {
            promise.reject("ACCESS_DENIED", "Bạn không có quyền thực hiện chức năng này");
            return;
        }

        executorService.execute(() -> {
            try {
                var file = new File(filePath);
                if (!file.exists()) {
                    promise.reject("FILE_NOT_FOUND", "File at path " + filePath + " does not exist");
                    return;
                }
                var resultUrl = mediaRepository.processCapturedImage(file);
                promise.resolve(resultUrl);
            } catch (Exception e) {
                promise.reject("UPLOAD_ERROR", e.getMessage(), e);
            }
        });
    }

    @ReactMethod
    public void performOcr(final String imageUrl, final Promise promise) {
        if (!permissionManager.canAccessFullFeatures()) {
            promise.reject("ACCESS_DENIED", "Bạn không có quyền thực hiện chức năng này");
            return;
        }

        executorService.execute(() -> {
            try {
                var ocrResult = mediaRepository.performOcr(imageUrl);
                promise.resolve(ocrResult);
            } catch (Exception e) {
                promise.reject("OCR_ERROR", e.getMessage(), e);
            }
        });
    }
}

File: Capstone-Project-dev/Capstone-Project-dev/fe/mobile/ios/ReactNative/Modules/Media/MediaBridgeModule.swift
import Foundation
import React

@objc(MediaModule)
class MediaModule: NSObject, RCTBridgeModule {
    static func moduleName() -> String! {
        return "MediaModule"
    }

    static func requiresMainQueueSetup() -> Bool {
        return false
    }

    private let mediaRepository = MediaRepository.shared
    private let permissionManager = PermissionManager.shared

    @objc(uploadCapturedImage:resolver:rejecter:)
    func uploadCapturedImage(_ filePath: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        if !permissionManager.canAccessFullFeatures() {
            reject("ACCESS_DENIED", "Bạn không có quyền thực hiện chức năng này", nil)
            return
        }

        mediaRepository.processCapturedImage(filePath: filePath) { resultUrl, error in
            if let error = error {
                reject("UPLOAD_ERROR", error.localizedDescription, error)
            } else {
                resolve(resultUrl)
            }
        }
    }

    @objc(performOcr:resolver:rejecter:)
    func performOcr(_ imageUrl: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        if !permissionManager.canAccessFullFeatures() {
            reject("ACCESS_DENIED", "Bạn không có quyền thực hiện chức năng này", nil)
            return
        }

        mediaRepository.performOcr(imageUrl: imageUrl) { ocrResult, error in
            if let error = error {
                reject("OCR_ERROR", error.localizedDescription, error)
            } else {
                resolve(ocrResult)
            }
        }
    }
}
